### PR TITLE
161 pagination plus 100 alternative names rebased

### PR DESCRIPTION
**Pull request by clarkepeterf**

### CODE REVIEW

File: src/main/ml-modules/root/lib/facetsLib.mjs [hunk 1]
```diff
@@ -16,72 +16,93 @@ import { BadRequestError } from './mlErrorsLib.mjs';
 import { FACETS_VIA_SEARCH_CONFIG } from '../config/facetsViaSearchConfig.mjs';
 import { facetToScopeAndTermName } from '../utils/searchTermUtils.mjs';
 
-// A safeguard from a current or future viaSearch facet request from pulling too many documents from disk.
-// Not a concern for the initial two facets but what about the next one?  As opposed to facets that pull
-// their values from a field, enabling a request to pull an unlimited number of documents from disk would
-// be bad.
-const MAXIMUM_FACET_VALUE_VIA_SEARCH_DOCUMENTS = 100;
```
Comment (brent-hartwig) [inline src/main/ml-modules/root/lib/facetsLib.mjs:23]: @clarkepeterf, can you find a place within this lib to include history for TTFU.  Perhaps at the top given the changes are sprinkled throughout.  Something similar to:

Two changes were made to restore system stability during performance tests: a) requests are now paginated and b) reduced the ratio between facet value and response body size.  For details, see #160 (non-semantic facet pagination), #161 (semantic facet pagination and facet response body change), and #162 (performance test).

Please also include rationale for the two page length limits.

File: src/main/ml-modules/root/lib/facetsLib.mjs [hunk 2]
```diff
@@ -16,75 +16,96 @@ import { BadRequestError } from './mlErrorsLib.mjs';
...
 
-// A safeguard from a current or future viaSearch facet request from pulling too many documents from disk.
-// Not a concern for the initial two facets but what about the next one?  As opposed to facets that pull
-// their values from a field, enabling a request to pull an unlimited number of documents from disk would
-// be bad.
-const MAXIMUM_FACET_VALUE_VIA_SEARCH_DOCUMENTS = 100;
+// Pagination constants
+const DEFAULT_PAGE = 1;
+const DEFAULT_PAGE_LENGTH = 20;
+const MAXIMUM_NON_SEMANTIC_PAGE_LENGTH = 10000;
```
Comment (brent-hartwig) [inline src/main/ml-modules/root/lib/facetsLib.mjs:22]: `MAXIMUM_SEMANTIC_PAGE_LENGTH` is presently used in two contexts:

1. Maximum page length, as the variable name convey.
2. Maximum number of results when searching for the facet values.

As such, one can never retrieve the 101st value for a semantic facet.  This was the case before pagination.  I'll update the API documentation to make this clear.

Comment (brent-hartwig) [reply src/main/ml-modules/root/lib/facetsLib.mjs:22]: Done

File: src/main/ml-modules/root/lib/facetsLib.mjs (hunk 1 above)
Comment (clarkepeterf) [reply src/main/ml-modules/root/lib/facetsLib.mjs:23]: @brent-hartwig I've added a comment explaining rationale. Mind reviewing?

**Pull request merged**